### PR TITLE
fix: TypeError in Validation is_unique/is_not_unique

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See https://forum.codeigniter.com/showthread.php?tid=86107

- fix TypeError in `Rules::is_unique()` and `is_not_unique()` in Strict Rules
- refactor tests
  - move the test logic to `StrictRules\DatabaseRelatedRulesTest` because Strict Rules are now default

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by michalsn — APPROVED** [review]

**Review by ddevsr — APPROVED** [review]

**Comment by jlopes90** [comment]
I already tested it and it works.

File: system/Validation/Rules.php [hunk 1]
```diff
@@ -86,7 +86,11 @@ public function greater_than_equal_to(?string $str, string $min): bool
     public function is_not_unique(?string $str, string $field, array $data): bool
```
Comment (jlopes90) [inline system/Validation/Rules.php:86]: I was trying something else, I removed "?string" and it worked.

`public function is_not_unique($str, string $field, array $data): bool`

**Comment by jlopes90** [comment]
I'm using "codeIgniter 4 app starter", how to get this to update or change this branch to vendor/codeigniter4/framework or something?

I tested was to copy the files and replace.

**Comment by kenjis** [comment]
@jlopes90 Thank you for testing.

See https://github.com/codeigniter4/shield/pull/195#issuecomment-1374456938
This is an example in Shield, but all you need to do is
1. add the repository URL in `repositories`
2. change the version to the branch (`dev-fix-Validtaion-is_unique` in this case)

**Pull request merged**